src/main/java/com/exadel/discount/repository/FavoriteRepository.java

```diff
@@ -32,4 +34,17 @@
     @EntityGraph(attributePaths = {"discount", "discount.category", "discount.vendorLocations", "discount.tags",
             "discount.vendor", "discount.vendorLocations.city", "discount.vendorLocations.city.country"})
     List<Favorite> findAllByIdIn(Iterable<UUID> ids, Sort sort);
+
+    @Modifying
+    void deleteFavoriteByDiscountIdAndUserEmail(@Param("discountId") UUID discountId,
+                                                @Param("userEmail") String userEmail);
+
+
+    @EntityGraph(attributePaths = {"user", "discount"})
+    Optional<Favorite> findByDiscountIdAndAndUserEmail(@Param("discountId") UUID discountId,
+                                                       @Param("userEmail") String userEmail);
+
+    @EntityGraph(attributePaths = {"user", "discount"})
+    boolean existsFavoriteByDiscountIdAndAndUserEmail(@Param("discountId") UUID discountId,
+                                                      @Param("userEmail") String userEmail);
```

Comment:
Just existsByDiscountIdAndUserEmail, you have typo in naming and you don't need EntityGraph here

Reply:
Done